bluesky/tests/test_external_assets_and_paging.py

```diff
@@ -1,479 +1,614 @@
-        )
-        for name in chain(["primary"], range(16))
-    ]
 
-    return partial_event_pages
+def collect_one_pv(self):
+    """Event producing value of a single pv"""
+    yield PartialEvent(data={"pv": 0}, timestamps={"pv": 100})
 
 
-def collect_events_all_detectors(self) -> SyncOrAsync[Iterator[PartialEvent]]:
-    def timestamps(x):
-        return {"det1": x + 0.1, "det2": x + 0.2, "det3": x + 0.3}
+class MultiKeyOldCollectable(Named, EventCollectable):
+    """Buggy Device that should fail describe_collect as it makes 2 streams with the same data_key"""
 
-    def data(x):
-        return {"det1": x, "det2": x*2, "det3": x*3}
+    describe_collect = describe_collect_two_streams_same_names
+    collect = collect_one_pv
+
+
+class OldPvAndDatumCollectable(Named, EventCollectable, WritesExternalAssets):
+    """Old style Device that produces 2 streams with PVs and 1 stream with Datum"""
+
+    describe_collect = merge_dicts(describe_collect_old_pv, describe_collect_old_datum)
+    collect = merge_iterators(collect_old_pv, collect_old_datum)
+    collect_asset_docs = collect_asset_docs_datum
+
+
+def describe_collect_pv(self: Named) -> Dict[str, Dict[str, DataKey]]:
+    """New style describe collect with 2 PVs"""
+    return {
+        f"{self.name}-{pv}": DataKey(source="pv", dtype="number", shape=[])
+        for pv in ["pv1", "pv2"]
+    }
 
-    partial_events = [
-        PartialEvent(
-            timestamps=timestamps(x), data=data(x), time=float(x)
+
+def collect_pv(self: Named) -> Iterator[PartialEvent]:
+    """Collect for the new style describe collect of 2 data keys from PVs"""
+    for i in range(2):
+        yield PartialEvent(
+            data={f"{self.name}-{pv}": i for pv in ["pv1", "pv2"]},
+            timestamps={f"{self.name}-{pv}": 100 + i for pv in ["pv1", "pv2"]},
         )
-        for x in range(16)
+
+
+def collect_pages_pv(self: Named) -> Iterator[PartialEvent]:
+    """Same as collect_pv but in EventPage form"""
+    yield PartialEvent(
+        data={f"{self.name}-{pv}": [0, 1] for pv in ["pv1", "pv2"]},
+        timestamps={f"{self.name}-{pv}": [100, 101] for pv in ["pv1", "pv2"]},
+    )
+
+
+class PvCollectable(Named, EventCollectable):
+    """Produces events with 2 data keys backed by PVs"""
+
+    describe_collect = describe_collect_pv
+    collect = collect_pv
+
+
+class PvPageCollectable(Named, EventPageCollectable):
+    """Produces event pages with 2 data keys backed by PVs"""
+
+    describe_collect = describe_collect_pv
+    collect_pages = collect_pages_pv
+
+
+class StreamDatumReadableCollectable(Named, Readable, Collectable, WritesStreamAssets):
+    """Produces no events, but only StreamResources for 2 data keys and can be read or collected"""
+
+    describe = describe_collect = describe_stream_datum
+    read = read_empty
+    collect_asset_docs = collect_asset_docs_stream_datum
+    get_index = get_index
+
+
+def test_datum_readable_counts(RE):
```

Comment:
```suggestion
def test_datum_readable_counts(RE):
    "Test that count-ing a datum-producing device results in expected documents."
```

Reply:
@abbiemery I can't remember what pytest does when you add a docstring to a test, please can you check that it doesn't hide the test name in the terminal...

If the output is still fine then all these are fine to go in